plugins/modules/vault_list.py

```diff
@@ -38,7 +38,7 @@
 - name: List kv2 secrets from Vault via the remote host with userpass auth
   community.hashi_vault.vault_list:
     url: https://vault:8201
-    path: secret
+    path: secret/metadata/path
```

Comment:
I got it wrong to start with - the path for kv2 needs to be `mount_point/metadata` so `secret/metadata` in this example. 
Both `secret` (i.e. the mount_point) and `secret/metadata/secret_name` fail for the list operation.

I will update in following commit.

Reply:
Done in https://github.com/ansible-collections/community.hashi_vault/pull/343/commits/857568ab7e72c9826cf56b5bb3691a4d950433d3